BDArmory/FX/ExplosionFX.cs

```diff
@@ -628,28 +644,45 @@ private void ExecuteBuildingBlastEvent(BuildingBlastHitEvent eventToExecute)
             //TODO: Review if the damage is sensible after so many changes
             //buildings
             DestructibleBuilding building = eventToExecute.Building;
-            building.damageDecay = 600f;
+            //building.damageDecay = 600f;
 
-            if (building)
+            if(building && building.IsIntact)
             {
                 var distanceFactor = Mathf.Clamp01((Range - eventToExecute.Distance) / Range);
-                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW * Power * distanceFactor;
-
-                damageToBuilding *= 2f;
+                float blastMod = 1;
+                switch (ExplosionSource)
+                {
+                    case ExplosionSourceType.Bullet:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW;
+                        break;
+                    case ExplosionSourceType.Rocket:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_ROCKET;
+                        break;
+                    case ExplosionSourceType.Missile:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_MISSILE;
+                        break;
+                    case ExplosionSourceType.BattleDamage:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BATTLE_DAMAGE;
+                        break;
+                }
+                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * blastMod * (Power * distanceFactor);
+                damageToBuilding /= 2;
                 damageToBuilding *= BDArmorySettings.BUILDING_DMG_MULTIPLIER;
+                //building.AddDamage(damageToBuilding); 
+                building.FacilityDamageFraction += damageToBuilding;
```

Comment:
Despite its name, FacilityDamageFraction doesn't top out a 1, or 100, and nothing seems to happen as it accumulates a value far in excess of either, so it works well as a damage tracker.
It also doesn't gate damage below a threshold.